src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeGenerator.java

```diff
@@ -154,38 +435,622 @@ private static void generateFromBody(Body body, CodeBuilder mv, ConversionContex
                 continue;
             }
 
+            c.setCurrentBlock(b);
             Label blockLabel = c.getLabel(b);
-            mv.labelBinding(blockLabel);
+            cob.labelBinding(blockLabel);
 
-            List<Op> ops = b.ops();
-            for (int i = 0; i < ops.size() - 1; i++) {
-                Op op = ops.get(i);
-                if (op instanceof BytecodeInstructionOps.InstructionOp inst) {
-                    inst.apply(mv, c);
-                } else if (op instanceof BytecodeInstructionOps.ControlInstructionOp inst) {
-                    inst.apply(mv, c);
+            // Assign slots to block arguments
+            b.parameters().forEach(c::getOrAssignSlot);
```

Comment:
This has the same problem as the one #8 fixes. You can probably include the changes from that PR here directly.